### PR TITLE
feat: navbar and react router page updates

### LINKED ISSUE #16: The ability to transfer to any main page using navigation bar throughout website pages
As a website user I would like the ability to move between different pages through a navigation bar.

Constraints:
- Size of navigation bar must dynamically change with window size.

Acceptance Criteria:
1. The navigation bar must appear at the top of every page.
2. The navigation bar must include links to the following pages in a dynamic format(referring to the session variable):
    a. Login page -> Profile page.
    b. Signup page -> Logout page

Tasks:
- [x] Megan - frontend: create blank pages for each of the new pages listed in acceptance criteria 2 (b, c, d, logout page) along with the search results, recipe, followers, following and edit profile pages.
- [x] Megan - frontend: add navigation bar to every page.
- [x] Megan - backend: add routes to each of the pages created in acceptance criteria 2.
- [x] Isam - frontend: create fully functional buttons listed in acceptance criteria 2.
- [x] Isam - frontend: connect to the backend to make sure every function works
- [x] Isam - backend: forward session variable to the frontend (specifically the navigation page) upon page change.

### PR DESCRIPTION
**Pull request by bdk99**
This creates the search results, recipe, followers, following and edit profile pages. This also includes the routes for each of the new pages. The navigation bar is added to and includes the Ingredient search page, Home page, Profile page, Login page, Logout page, and Signup page. 
Closes #16

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,27 +1,42 @@
...
 import { BrowserRouter as Router, Routes, Route} from 'react-router-dom';
 import { ReactSession } from 'react-client-session';
 
+//Imports pages to allow BrowserRouter to access them
 import Home from './pages';
 import Profile from './pages/profile';
 import SignUp from './pages/signup';
 import Login from './pages/login';
 import PasswordReset from './pages/passwordReset';
+import Logout from './pages/logout';
```
Comment (bdk99) [inline client/src/App.js]: Do not import logout page here.  Ryan and I already handled this on the view-profile PBI

File: client/src/App.js [hunk 2]
```diff
@@ -1,27 +1,42 @@
...
                 <Route path='/profile' element={<Profile/>} />
                 <Route path='/login' element={<Login/>} />
                 <Route path='/passwordReset' element={<PasswordReset/>} />
+                <Route path='/logout' element={<Logout/>} />
+                <Route path='/seacrhResults' element={<SeacrhResults/>} />
+                <Route path='/recipe' element={<Recipe/>} />
+                <Route path='/followers' element={<Followers/>} />
+                <Route path='/following' element={<Following/>} />
+                <Route path='/editProfile' element={<EditProfile/>} />
+                <Route path='/ingredientSeacrh' element={<IngredientSearch/>} />
```
Comment (bdk99) [inline client/src/App.js]: Fix spelling of ingredientSeacrh on element and path

File: client/src/components/Navbar/index.js [hunk 3]
```diff
@@ -1,26 +1,40 @@
...
-		<li>
-			<Link to="/profile">Profile</Link>
-		</li>
+			<li>
+				<Link to="/">Home</Link>
+			</li>
+      <li>
+        <Link to="/ingredientSearch">Ingredient Search</Link>
+      </li>
+			{/*Makes the navbar dynamic, displays the login and signup buttons if the user is not signed in*/}
```
Comment (bdk99) [inline client/src/components/Navbar/index.js:19]: Fix formatting/indentation

File: client/src/App.js [hunk 4]
```diff
@@ -1,27 +1,42 @@
...
 
+//Imports pages to allow BrowserRouter to access them
 import Home from './pages';
 import Profile from './pages/profile';
 import SignUp from './pages/signup';
 import Login from './pages/login';
 import PasswordReset from './pages/passwordReset';
+import Logout from './pages/logout';
+import EditProfile from './pages/editProfile';
+import SeacrhResults from './pages/searchResults';
```
Comment (bdk99) [inline client/src/App.js]: fix spelling of SeacrhResults

File: client/src/App.js [hunk 5]
```diff
@@ -1,27 +1,42 @@
...
         return (
             <Router>
-            <Navbar />
             <Routes>
                 <Route path='/' exact element={<Home />} />
                 <Route path='/signup' element={<SignUp/>} />
                 <Route path='/profile' element={<Profile/>} />
                 <Route path='/login' element={<Login/>} />
                 <Route path='/passwordReset' element={<PasswordReset/>} />
+                <Route path='/logout' element={<Logout/>} />
```
Comment (bdk99) [inline client/src/App.js]: Remove logout route

File: client/src/App.js [hunk 6]
```diff
@@ -1,27 +1,42 @@
...
+                <Route path='/seacrhResults' element={<SeacrhResults/>} />
```
Comment (bdk99) [inline client/src/App.js]: Fix spelling of seacrhResults for path and element

File: client/src/pages/editProfile.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
```
Comment (bdk99) [inline client/src/pages/editProfile.js:3]: Add navbar import statement here

import Navbar from '../components/Navbar';

File: client/src/pages/followers.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
```
Comment (bdk99) [inline client/src/pages/followers.js:3]: Add navbar import statement here

import Navbar from '../components/Navbar';

File: client/src/pages/following.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
```
Comment (bdk99) [inline client/src/pages/following.js:3]: Add navbar import statement here

import Navbar from '../components/Navbar';

File: client/src/pages/ingredientSearch.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
```
Comment (bdk99) [inline client/src/pages/ingredientSearch.js]: Add navbar import statement here

import Navbar from '../components/Navbar';

File: client/src/pages/logout.js [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+  
+const Logout = () => {
+  return (
+    <div>
+      <h1>This is the logout page</h1>
+    </div>
+  );
+};
+  
+export default Logout;
```
Comment (bdk99) [inline client/src/pages/logout.js]: Remove this file entirely

File: client/src/pages/recipe.js [hunk 12]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
```
Comment (bdk99) [inline client/src/pages/recipe.js:3]: Add navbar import statement here

import Navbar from '../components/Navbar';

File: client/src/pages/searchResults.js [hunk 13]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
```
Comment (bdk99) [inline client/src/pages/searchResults.js:3]: Add navbar import statement here

import Navbar from '../components/Navbar';

File: client/src/pages/editProfile.js [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+const EditProfile = () => {
+return (
+	<div>
+	<h1>Edit Profile Page</h1>
```
Comment (bdk99) [inline client/src/pages/editProfile.js]: Add navbar import statement between lines 5 and 6

File: client/src/pages/followers.js [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+const Followers = () => {
+return (
+	<div>
+	<h1>Followers Page</h1>
```
Comment (bdk99) [inline client/src/pages/followers.js]: Add navbar import statement between lines 5 and 6

File: client/src/pages/following.js [hunk 16]
```diff
@@ -0,0 +1,11 @@
...
+const Following = () => {
+return (
+	<div>
+	<h1>Following Page</h1>
```
Comment (bdk99) [inline client/src/pages/following.js]: Add navbar import statement between lines 5 and 6

File: client/src/pages/ingredientSearch.js [hunk 17]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+  
+const IngredientSearch = () => {
+  return (
+    <div>
+      <h1>This is the ingredient search page</h1>
```
Comment (bdk99) [inline client/src/pages/ingredientSearch.js]: Add navbar import statement between lines 5 and 6

File: client/src/pages/recipe.js [hunk 18]
```diff
@@ -0,0 +1,11 @@
...
+const Recipe = () => {
+return (
+	<div>
+	<h1>Recipe Page</h1>
```
Comment (bdk99) [inline client/src/pages/recipe.js]: Add navbar import statement between lines 5 and 6

File: client/src/pages/searchResults.js [hunk 19]
```diff
@@ -0,0 +1,11 @@
...
+const SeacrhResults = () => {
+return (
+	<div>
+	<h1>Search Results Page</h1>
```
Comment (bdk99) [inline client/src/pages/searchResults.js]: Add navbar import statement between lines 5 and 6

**Comment by RyanCornwall** [comment]
Looks good.

**Pull request merged**